pkg/kubelet/pod_manager.go

```diff
@@ -171,6 +172,15 @@ func (pm *basicPodManager) getAllPods() []*api.Pod {
 	return append(podsMapToPods(pm.podByUID), podsMapToPods(pm.mirrorPodByUID)...)
 }
 
+// GetPodByUID provides the (non-mirror) pod that matches pod UID, as well as
+// whether the pod is found.
+func (pm *basicPodManager) GetPodByUID(uid types.UID) (*api.Pod, bool) {
+	pm.lock.RLock()
+	defer pm.lock.RUnlock()
+	pod, ok := pm.podByUID[uid]
```

Comment:
We can. I think we used to return an `api.Pod` object (not a pointer), so it wasn't an option. Now that every function in this file has the same return format, we may want to change them in a separate pr.